app/services/work_packages/set_attributes_service.rb

```diff
@@ -292,8 +354,11 @@ def children_duration
     max = max_child_date
 
     return unless max
+    days.duration(min_child_date, max_child_date)
+  end
 
-    max - min_child_date + 1
+  def days
+    WorkPackages::Shared::Days.for(work_package)
   end
 
   def max_child_date
```

Comment:
I know this is out of scope for the PR, but do you think it makes sense to change this method body to the following?

```ruby
work_package.children.flat_map { |wp| [wp.start_date, wp.due_date] }.compact.max
```

Reply:
yep, I thought of something similar too when I saw it. I'll change it.

Reply:
Done in [b9bc90e](https://github.com/opf/openproject/pull/11177/commits/b9bc90e9c7047409064b44d24945696a736803bf)